nutanix/resource_nutanix_ndb_profiles.go

```diff
@@ -0,0 +1,1280 @@
+		d.Set("assoc_databases", nil)
+	}
+
+	if resp.Nxclusterid != nil {
+		d.Set("nx_cluster_id", resp.Nxclusterid)
+	} else {
+		d.Set("nx_cluster_id", nil)
+	}
+
+	if resp.Clusteravailability != nil {
+		d.Set("cluster_availability", flattenClusterAvailability(resp.Clusteravailability))
+	} else {
+		d.Set("cluster_availability", nil)
+	}
+
+	return nil
+}
+
+func resourceNutanixNDBProfileUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	req := &era.ProfileRequest{}
+
+	netReq := &era.UpdateProfileRequest{}
+
+	res, err := conn.Service.GetProfiles(ctx, "", "", d.Id(), "")
+	if err != nil {
+		diag.FromErr(err)
+	}
+
+	if res != nil {
+		netReq.Name = res.Name
+		netReq.Description = res.Description
+		req.Properties = res.Versions[0].Properties
+	}
+
+	if pub, ok := d.GetOk("published"); ok {
+		req.Published = pub.(bool)
+	}
+
+	if d.HasChange("name") {
+		netReq.Name = utils.StringPtr(d.Get("name").(string))
+		// update version name as well
+		versionName := d.Get("name").(string)
+		updateVersionName := versionName + " " + " (1.0)"
+		req.Name = utils.StringPtr(updateVersionName)
+	}
+
+	if d.HasChange("description") {
+		netReq.Description = utils.StringPtr(d.Get("description").(string))
+		req.Description = utils.StringPtr(d.Get("description").(string))
+	}
+
+	if d.HasChange("compute_profile") {
+		req.Properties = buildComputeProfileRequest(d.Get("compute_profile"))
+	}
+
+	if d.HasChange("network_profile") {
+		nps := d.Get("network_profile").([]interface{})
+
+		for _, v := range nps {
+			val := v.(map[string]interface{})
+
+			if ps, ok := val["postgres_database"]; ok {
+				req.Properties = expandNetworkProfileProperties(ctx, meta, ps.([]interface{}))
+			}
+
+			if cls, ok := val["version_cluster_association"]; ok {
+				clster := cls.([]interface{})
+				out := make([]*era.VersionClusterAssociation, len(clster))
+				for _, v := range clster {
+					val := v.(map[string]interface{})
+
+					if p1, ok1 := val["nx_cluster_id"]; ok1 {
+						out = append(out, &era.VersionClusterAssociation{
+							NxClusterID: utils.StringPtr(p1.(string)),
+						})
+					}
+				}
+				req.VersionClusterAssociation = out
+			}
+		}
+	}
+
+	if d.HasChange("database_parameter_profile") {
+		req.Properties = buildDatabaseProfileProperties(d.Get("database_parameter_profile").([]interface{}))
+	}
+
+	if d.HasChange("software_profile") {
+		splist := d.Get("software_profile").([]interface{})
+
+		for _, v := range splist {
+			val := v.(map[string]interface{})
+
+			if ps, ok := val["postgres_database"]; ok {
+				req.Properties = expandSoftwareProfileProp(ps.([]interface{}))
+			}
+
+			if ac, ok1 := d.GetOk("available_cluster_ids"); ok1 {
+				st := ac.([]interface{})
+				sublist := make([]*string, len(st))
+
+				for a := range st {
+					sublist[a] = utils.StringPtr(st[a].(string))
+				}
+				req.AvailableClusterIds = sublist
+			}
+		}
+	}
+
+	versionID := res.Versions[0].ID
```

Comment:
Versions[0] is not always the case of all kind of profiles right ? Software profile can have >1 profile versions